packages/twenty-front/src/modules/ui/field/input/components/RatingInput.tsx

```diff
@@ -31,24 +32,40 @@ export const RatingInput = ({
   readonly,
 }: RatingInputProps) => {
   const { theme } = useContext(ThemeContext);
+  const clearField = useClearField();
 
   const activeColor = theme.font.color.secondary;
   const inactiveColor = theme.background.quaternary;
 
-  const [hoveredValue, setHoveredValue] = useState<FieldRatingValue | null>(
-    null,
-  );
+  const [hoveredValue, setHoveredValue] = useState<FieldRatingValue>(null);
+
   const currentValue = hoveredValue ?? value;
 
-  const selectedIndex = RATING_VALUES.indexOf(currentValue);
+  const selectedIndex = currentValue !== null ? RATING_VALUES.indexOf(currentValue) : -1;
+
+  const previousRating = useRef<FieldRatingValue>(null);
+
+  const handleClick = (value: FieldRatingValue) => {
+    if (readonly) return;
+    if (previousRating.current === currentValue) {
+      setHoveredValue(null);
+      clearField();
+    } else {
+      onChange?.(value);
+    }
+  };
+
+  useEffect(() => {
```

Comment:
Can you try getting rid of the effect? https://react.dev/learn/you-might-not-need-an-effect
Looks good otherwise, thanks!

Reply:
Sure I'll check this

Reply:
@FelixMalfait I have removed the useEffect